cpp/arcticdb/util/configs_map.hpp

```diff
@@ -21,11 +21,11 @@ using namespace arcticdb::proto::config;
 
 class ConfigsMap {
 public:
-    static std::shared_ptr<ConfigsMap> instance_;
-    static std::once_flag init_flag_;
-
     static void init();
-    static std::shared_ptr<ConfigsMap> instance();
+    static std::shared_ptr<ConfigsMap> instance() { 
+        static auto instance_ = std::make_shared<ConfigsMap>();
+        return instance_;
+    }
 
     ConfigsMap() :
         mutex_(std::make_unique<std::mutex>()) {
```

Comment:
I don't understand what good is this mutex here (it is not used anywhere AFAIC) and why is it wrapped in a pointer. Anyone more familiar with this please give some insight. Otherwise, I think I will just add a normal mutex (not ptr) and add it to all the operations here